# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.h
# Commit: 920da79f43fc
# Full Hash: 920da79f43fcf5802ebd5eb2bf610a18e821dacc
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-19 03:53:34
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r 98d7f9e4839f -r 920da79f43fc js/src/wasm/WasmTypes.h
--- a/js/src/wasm/WasmTypes.h	Wed Mar 18 17:12:14 2020 +0000
+++ b/js/src/wasm/WasmTypes.h	Wed Mar 18 16:36:45 2020 +0000
@@ -1086,6 +1086,16 @@
     }
     return false;
   }
+  // Entry from JS to wasm is currently unimplemented for functions that return
+  // multiple values.
+  bool temporarilyUnsupportedResultCountForEntry() const {
+    return results().length() > 1;
+  }
+  // Calls out from wasm to JS that return multiple values is currently
+  // unsupported.
+  bool temporarilyUnsupportedResultCountForExit() const {
+    return results().length() > 1;
+  }
   // For JS->wasm jit entries, AnyRef parameters and returns are allowed,
   // as are all reference types apart from TypeIndex.
   bool temporarilyUnsupportedReftypeForEntry() const {